research-hub-web/src/app/components/casestudys/case-study/case-study.component.ts

```diff
@@ -0,0 +1,108 @@
+import { Component, OnDestroy, OnInit, Type } from '@angular/core';
+import { ActivatedRoute, Router } from '@angular/router';
+import { CaseStudy, GetCaseStudyBySlugGQL } from '@graphql/schema';
+import { BodyMediaService } from '@services/body-media.service';
+import { PageTitleService } from '@services/page-title.service';
+import { MarkRenderer, NodeRenderer } from 'ngx-contentful-rich-text';
+import { Observable, of, Subscription, throwError } from 'rxjs';
+import { map, mergeMap, switchMap } from 'rxjs/operators';
+import supportsWebP from 'supports-webp';
+
+
+@Component({
+  selector: 'app-case-study',
+  templateUrl: './case-study.component.html',
+  styleUrls: ['./case-study.component.scss']
+})
+export class CaseStudyComponent implements OnInit, OnDestroy {
+  public nodeRenderers: Record<string, Type<NodeRenderer>>;
+  public markRenderers: Record<string, Type<MarkRenderer>>;
+
+  private subscriptions = new Subscription();
+
+  public bannerTextStyling = 'color: white; text-shadow: 0px 0px 8px #333333;';
+  public caseStudy: CaseStudy;
+  public supportsWebp: Boolean;
+  public bannerImageUrl: string;
+
+  constructor(
+    public route: ActivatedRoute,
+    public getCaseStudyBySlugGQL: GetCaseStudyBySlugGQL,
+    public pageTitleService: PageTitleService,
+    public bodyMediaService: BodyMediaService,
+    public router: Router,
+  ) {
+    this.detectWebP();
+
+    this.nodeRenderers = this.bodyMediaService.nodeRenderers;
+    this.markRenderers = this.bodyMediaService.markRenderers;
+  }
+
+  detectWebP() {
+    supportsWebP.then(supported => {
+      this.supportsWebp = supported;
+    });
+  }
+
+  ngOnInit() {
+    this.subscriptions.add(this.route.params.pipe(
+      map((params) => {
+        return (params.slug || this.route.snapshot.data.slug) as string;
+      }),
+      switchMap((slug) => this.loadCaseStudy(slug))
+    ).subscribe({
+      next: (caseStudy: CaseStudy) => this.caseStudy = caseStudy,
+      error: (error) => {
+        console.error(error);
+        this.router.navigate(['error', 404]);
```

Comment:
Just realised we need to handle different types of errors here - there could also be auth errors or other unexpected errors, so just need to check and handle the two known cases (unauthenticated, and not found) and then have a third catch-all where we just log to the console.